foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/utils/JvmUtils.java

```diff
@@ -36,21 +42,50 @@ private JvmUtils() {
    * @return main class or null, never throw exception
    */
   public static Class<?> findMainClass() {
+    // 1.run with java -cp ......
+    //   command is main class and args
+    // 2.run with java -jar ......
+    //   command is jar file name and args
     String command = System.getProperty(SUN_JAVA_COMMAND);
     if (command == null || command.isEmpty()) {
       return null;
     }
 
-    // command is main class and args
-    String mainClass = command.trim().split(" ")[0];
+    String mainClassOrJar = command.trim().split(" ")[0];
+    String mainClass = readFromJar(mainClassOrJar);
+    if (mainClass == null || mainClass.isEmpty()) {
+      return null;
```

Comment:
We may need add a log message here also.

Reply:
already print in "readFromJar"

Reply:
OK